lib/ansible/module_utils/urls.py

```diff
@@ -302,6 +302,16 @@ def match_hostname(cert, hostname):
 
     HAS_MATCH_HOSTNAME = True
 
+if hasattr(socket, 'create_connection'):
+    create_connection = socket.create_connection
+else:
+    # Just a graceful fallback for py < 2.6, implementing old behaviour.
+    # Wouldn't work with IPv6, but it is unlikely that python 2.4 could be met
+    # on IPv6-only machine.
+    def create_connection(address):
+        sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        sock.connect(address)
+        return sock
```

Comment:
I don't have a huge preference here but I think @sivel's code is slightly better. We do use try: except (Ask forgiveness) quite a lot for compatibility between python versions.  Asking forgiveness seems to be the accepted idiomatice way to write python versus look before you leap.

Since this is for python-2.4 only it makes even more sense.  In the large picture there's only a few users running python-2.4 and we're going to be getting rid of support for it in less than a year.

Probably the major problem with the approach currently in the code is that it creates an ambiguity in what happens when create_connection() is called that isn't apparent when you are looking at the caller.  This can be worked around by adding the new function under a different name and then putting the conditional into the calling code to select which of the two functions to call or putting the condition and the implementing code into the caller (The latter repeats itself once but for this specific case is probably okay -- it adds less lines of code which often indicates that the function can be considered too trivial to warrant being moved outside the caller).  That will be slower for the common case of operating on non-python-2.4 (exceptions are faster than conditionals when following the happy path) but may be a good compromise.  @sivel is the exception vs conditional the major thing you're looking at or the obfuscation of create_connection's meaning and the relative burden between caller and called?